### PR TITLE
Pair-Programming: Add devise

### PR DESCRIPTION
**Pull request by PeteUgwu**
## BlogApp - Add devise

In this Milestone, We:
- [ ] Removed `current_user` method in `ApplicationController`, since devise provided us with one.
- [ ] [Installed and setup devise](https://github.com/heartcombo/devise#getting-started) so it/a:
    - Can register a new user.
    - User logs in with a combination of email and password.
    - Hashed password should be stored in the database.
    - Asks for confirmation by email.
    - Can reset password.
- [ ] [Modified the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and our styling.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,5 +1,12 @@
 class ApplicationController < ActionController::Base
```
Comment (peterrobert) [inline app/controllers/application_controller.rb:1]: Adding this code ` before_action :authenticate_user!` to this controller will be great. The reason is, although you have set up sign_up and sign_in correctly and everything is working as expected, your application is misbehaving. For example, a user who is not authenticated can access your application's root page. Also when you sign out a user it doesn't really work as expected. Kindly add that line to your code and test your application and you will notice the difference

**Review by peterrobert — CHANGES_REQUESTED** [review]
Hi  @PeteUgwu  and @bernabasy , 

Very nice project. I love that you have added device to your project 👏🏿 . However, there are a few changes to be made kindly check below ⬇️

### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️
[ ] Great work ensuring the designs for the login and sign up match the wireflames✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (PeteUgwu) [reply app/controllers/application_controller.rb:1]: This is so helpful. `Updated!`.
Thanks, @peterrobert

Comment (peterrobert) [reply app/controllers/application_controller.rb:1]: welcome anytime @PeteUgwu  and @bernabasy

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @PeteUgwu, @bernabasy and @asifaminisonline,

✔️ Authentication of user

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**